traffic_portal/app/src/common/api/CdniService.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var CdniService = function($http, ENV, messageModel) {
+
+	this.getCdniConfigRequests = function() {
+		return $http.get(ENV.api.unstable + 'OC/CI/configuration/requests').then(
+			function (result) {
+				return result.data.response;
+				},
```

Comment:
this isn't something you need to change, but just for future reference our webpack version has been updated so we can use ES6 features in TP now. And in case that doesn't mean anything to you, it means we can use "fat arrow functions" to express these inline functions more succinctly with the syntax `(arg, list) => {some; statements;}`/`arg => {some; statements;}`/`(arg, list) => result`/`arg=>result`.

Like I said you don't have to change it, so I'm not gonna point out everywhere you could, but just to use this callback as an example, you could write the `getCdniConfigRequests` as such:
```javascript
this.getCdniConfigRequests = () => $http.get(`${ENV.api.unstable}OC/CI/configuration/requests`).then(
    result => result.data.response,
    err => {
        messageModel.setMessages(err.data.alerts, true);
        throw err;
    }
);
```
which I like a lot more than having to write out all the boilerplate of `function` and `return` and `{`/`}` pairs each time.

Reply:
this is really good to know! thats a lot cleaner!